test/infrastructure/docker/internal/provisioning/cloudinit/writefiles.go

```diff
@@ -133,9 +133,17 @@ func fixNodeRegistration(nodeRegistration *bootstrapv1.NodeRegistrationOptions,
 		nodeRegistration.KubeletExtraArgs = map[string]string{}
 	}
 
+	// Disable disk resource management by default.
+	// The kubelet will see the host disk that the inner container runtime
+	// is ultimately backed by and attempt to recover disk space. We don't want that.
+	// See https://github.com/kubernetes-sigs/cluster-api/issues/11856 for more details.
+	if _, ok := nodeRegistration.KubeletExtraArgs[""]; !ok {
```

Comment:
```suggestion
	if _, ok := nodeRegistration.KubeletExtraArgs["image-gc-high-threshold"]; !ok {
```
I think?

Reply:
Maybe worth a `defaultExtraArg` func at this point  (we have this same logic a bunch of times below)